### PR TITLE
zammad: 5.1.1 -> 5.3.1

### PR DESCRIPTION
**Pull request by 0xSbock**
###### Description of changes
- Bump package version
- Includes Security patches for https://zammad.com/en/advisories/zaa-2022-11
- Release notes: https://zammad.com/en/releases/5-3-1

###### Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [x] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [x] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [x] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [23.05 Release Notes (or backporting 22.11 Release notes)](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#generating-2305-release-notes)
  - [x] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

File: pkgs/applications/networking/misc/zammad/default.nix [hunk 1]
```diff
@@ -30,9 +30,9 @@ let
     patches = [ ./0001-nulldb.patch ];
 
     postPatch = ''
-      sed -i -e "s|ruby '2.7.4'|ruby '${ruby_2_7.version}'|" Gemfile
-      sed -i -e "s|ruby 2.7.4p191|ruby ${ruby_2_7.version}|" Gemfile.lock
-      sed -i -e "s|2.7.4|${ruby_2_7.version}|" .ruby-version
+      sed -i -e "s|ruby '3.0.4'|ruby '${ruby_3_0.version}'|" Gemfile
+      sed -i -e "s|ruby 3.0.4p208|ruby ${ruby_3_0.version}|" Gemfile.lock
+      sed -i -e "s|3.0.4|${ruby_3_0.version}|" .ruby-version
```
Comment (SuperSandro2000) [inline pkgs/applications/networking/misc/zammad/default.nix:35]: Please use wildcard to match this.

**Review by SuperSandro2000 — COMMENTED** [review]
Do we really need to add 20k lines of vendored things?

File: pkgs/applications/networking/misc/zammad/default.nix [hunk 2]
```diff
@@ -94,6 +94,8 @@ let
     yarnLock = ./yarn.lock;
     yarnNix = ./yarn.nix;
     packageJSON = ./package.json;
+    #FIXME: packageResolutions should be remove when https://github.com/NixOS/nixpkgs/pull/214952/ is merged
+    packageResolutions = (lib.importJSON ./package.json).resolutions;
```
[the remark this answers is not in the document]
Comment (0xSbock) [reply pkgs/applications/networking/misc/zammad/default.nix:98]: without the `packageResolutions` key this wouldn't build. i wanted to make it easier in the future to remove this line again if it isn't needed anymore. should i refer to the open issue in some other way?

**Comment by 0xSbock** [comment]
> Do we really need to add 20k lines of vendored things?

First of all thanks for the quick review! :)
As this is my first PR on `nixpkgs` i didn't want to change that much of the of the already provided package.
I will delete `./yarn.lock` as we can use the file provided by the upstream source.
After looking at other packages in `nixpkgs` which are using `mkYarnPackage`, deleting `yarn.nix` should work as well as long as the upstream `yarn.lock` file is present.
I don't see how i could get rid of `gemset.nix` and though :/

File: pkgs/applications/networking/misc/zammad/default.nix (hunk 2 above)
[the remark this answers is not in the document]
Comment (Radvendii) [reply pkgs/applications/networking/misc/zammad/default.nix:97]: I suspect if we do come up with a solution, we should reference it from that PR anyways, so people will be able to find it. And I don't see a better way of referencing it.

**Review by Radvendii — APPROVED** [review]

**Comment by SuperSandro2000** [comment]
please remove drafting when the eval no longer fails because of IFD

**Comment by Radvendii** [comment]
Wait, does *this* PR trigger IFD, or the PR referenced in the comment?

**Comment by Radvendii** [comment]
If this PR is triggering IFD, I would guess it's because of [this line getting removed](https://github.com/NixOS/nixpkgs/pull/217329/files#diff-631b0892eaefa8badbcde1930c91a15c2cdad23d9be72e1c89dbe4aef0531751L95), and we have to keep the local yarn.nix

**Comment by mweinelt** [comment]
Superseded by #231512

**Pull request closed without merging**